lua/ulx/modules/timed_gag.lua

```diff
@@ -0,0 +1,312 @@
+-- Constants
+local ULX_CATEGORY_NAME = "CFC"
+local GAGS_SQL_TABLE = "cfc_timed_gags"
+local INIT_WAIT_TIME = 1
+local GAG_CHECK_INTERVAL = 1
+
+local GaggedPlayers = {}
+
+local gagsInitialized = false
+
+local GAG_QUERIES = {}
+GAG_QUERIES['create_gag']   = "INSERT INTO %s(steam_id, expiration, reason) VALUES('%s', %d, '%s')"
+GAG_QUERIES['create_table'] = "CREATE TABLE %s(steam_id TEXT NOT NULL UNIQUE, expiration BIGINT, reason TEXT)"
+GAG_QUERIES['delete_gag']   = "DELETE FROM %s WHERE steam_id='%s'"
+GAG_QUERIES['retrieve_gag'] = "SELECT %s FROM %s WHERE steam_id='%s'"
+GAG_QUERIES['update_gag']   = "UPDATE %s SET expiration=%d,reason='%s' WHERE steam_id='%s'"
+
+
+-- HELPERS --
+
+local function IsValidPlayer(ply)
+    local playerIsValid = IsValid( ply ) and ply:IsPlayer()
+
+    return playerIsValid
+end
+
+local function GagPrint(msg)
+    print( "[CFC Timed Gag] " .. msg )
+end
+
+local function DatabaseGagPrint(action, succeeded, ply)
+    local result = ( succeeded == false ) and "FAILED" or "SUCCEEDED"
+
+    if IsValidPlayer( ply ) then
+        GagPrint( action .. " for '" .. ply:Nick() .. "' (" .. ply:SteamID() .. ") in Database " .. result )
+    else
+        GagPrint( action .. " " .. result )
+    end
+end
+
+-- END HELPERS --
+
+
+
+-- DATABASE OPERATIONS --
+
+local function SQLOperation(action, query, ply, isRead)
+    -- Use QueryValue if it's a read operation, otherwise just Query
+    local result = isRead and sql.QueryValue( query ) or sql.Query( query )
+
+    -- False is failure in both cases, but nil is only a fail for read operations
+    local failed = ( result == false ) or ( isRead and result == nil )
+    local succeeded = not failed
+
+    DatabaseGagPrint( action, succeeded, ply )
+
+    return result
+end
+
+local function SQLAction(action, query, ply)
+    return SQLOperation( action, query, ply, false )
```

Comment:
Does it make more sense to use `sql.SQLStr` here instead of where we call this function?

Reply:
Using it here breaks _everything._
The only thing in our queries determined by user input is the `reason` column, and that's what I've wrapped.